src/udp/socket.rs

```diff
@@ -861,15 +861,17 @@ where
         chooses_sent
     );
 
-    let msg = ctx.encrypt_msg(&HolePunchMsg::Choose);
+    let handle = handle.clone();
+    let encrypted = unwrap!(secure_serialise(&HolePunchMsg::Choose, &their_pk, &our_sk));
```

Comment:
Let's handle errors instead of panic?

Reply:
That definitely shouldn't produce an error though. You should always be able to serialize a message.

Reply:
Right, but we still rely on the underlying library and if for some reasons it returns error, we should be better off handling it. That's something we discussed with Spandan before and he was against unwrapping in this case. Also it leaves no questions for future developers why we have unwrap here while this should be an infallible operation.